src/components/ManageAccountsDrawer/components/ImportAccounts.tsx

```diff
@@ -176,37 +188,63 @@ const LoadingRow = ({ numRows }: { numRows: number }) => {
 }
 
 export const ImportAccounts = ({ chainId, onClose, isOpen }: ImportAccountsProps) => {
+  // State
+  const [isAutoDiscovering, setIsAutoDiscovering] = useState(true)
+  const [queryEnabled, setQueryEnabled] = useState(false)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [toggledAccountIds, setToggledAccountIds] = useState<Set<AccountId>>(new Set())
+
+  // Misc
   const translate = useTranslate()
   const dispatch = useAppDispatch()
   const queryClient = useQueryClient()
   const {
     state: { wallet, deviceId: walletDeviceId },
   } = useWallet()
-  const asset = useAppSelector(state => selectFeeAssetByChainId(state, chainId))
   const { isSnapInstalled } = useIsSnapInstalled()
   const isLedgerWallet = useMemo(() => wallet && isLedger(wallet), [wallet])
   const isMetaMaskMultichainWallet = useMemo(
     () => wallet instanceof MetaMaskMultiChainHDWallet,
     [wallet],
   )
+
+  // Selectors
+  const nftCollectionsById = useAppSelector(selectNftCollections)
+  const asset = useAppSelector(state => selectFeeAssetByChainId(state, chainId))
+
+  // Prefetch Portals account data, ish. At this point, we already have querydata for all *enabled* AccountIds,
+  // so this will really fetch it for the newly toggled ones
```

Comment:
Precisely what this says (read the following assuming you're within 5mn of opening the app, see comment above):

All accounts which are already fetch *will* have their Portals account data fetched already. This really only ensures that the next account index has its Portals account data prefetched on toggle on, so that clicking Done doesn't fire a XHR and slow things down.